### PR TITLE
Only allow immutable statics with #[linkage]

### PR DESCRIPTION
**Pull request by bjorn3**

### CODE REVIEW

File: compiler/rustc_codegen_ssa/src/codegen_attrs.rs [hunk 1]
```diff
@@ -327,6 +327,10 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, did: LocalDefId) -> CodegenFnAttrs {
                     } else {
                         codegen_fn_attrs.linkage = linkage;
                     }
+                    if tcx.is_mutable_static(did.into()) {
+                        tcx.dcx()
+                            .span_err(attr.span, "mutable statics are not allowed with #[linkage]");
```
Comment (compiler-errors) [inline compiler/rustc_codegen_ssa/src/codegen_attrs.rs:332]:
```suggestion
                            .span_err(attr.span, "mutable statics are not allowed with `#[linkage]`");
```

Also maybe worth while explaining *why* in a `note`, or linking to something that explains why.

Comment (bjorn3) [reply compiler/rustc_codegen_ssa/src/codegen_attrs.rs:332]: Done

**Comment by cjgillot** [comment]
Should this be considered as a breaking change that requires t-lang signoff, or as something that has never worked ever?

**Comment by bjorn3** [comment]
`#[linkage]` is an unstable feature. It seems like an oversight to me that `static mut` was not denied when used with `#[linkage]`.

**Comment by cjgillot** [comment]
Indeed.
@bors r+

**Pull request merged**